gemini/reasoning-engine/tutorial_graph_rag.ipynb

```diff
@@ -0,0 +1,1707 @@
+    "\n",
+    "    print(\n",
+    "        \"Created database {} on instance {}\".format(\n",
+    "            database.name,\n",
+    "            database_admin_api.instance_path(spanner_client.project, instance_id)\n",
+    "        )\n",
+    "    )\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "7mVK78UWt2OJ",
+    "outputId": "9e4e0725-fcc2-4071-b853-93486c262453",
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from google.cloud import spanner\n",
+    "create_database(GCP_PROJECT_ID, SPANNER_INSTANCE_ID, SPANNER_DATABASE_ID)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "wmZK7O3x56iN"
+   },
+   "source": [
+    "Now that the Spanner instance, database and tables are created. We will loop through the graph nodes and relationship in the knowledge graph and store them in temperary arrays to insert them into the Spanner database."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "28RVIpzndL3c",
+    "outputId": "ac74f4ce-eb50-493a-b67a-f9fbaefd6171",
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[['Person', 'Lawrence Edward Page'], ['Organization', 'Google'], ['Organization', 'Opener'], ['Person', 'Sergey Brin'], ['Organization', 'Alphabet Inc.'], ['Person', 'Maria Montessori'], ['Asset', '$156 billion'], ['Organization', 'Kitty Hawk'], ['Asset', '$145.2 billion']]\n",
+      "[['NET_WORTH', 'Person', 'Lawrence Edward Page', 'Asset', '$156 billion'], ['NET_WORTH', 'Person', 'Lawrence Edward Page', 'Asset', '$145.2 billion'], ['WORKED_AT', 'Person', 'Lawrence Edward Page', 'Organization', 'Google'], ['WORKED_AT', 'Person', 'Lawrence Edward Page', 'Organization', 'Alphabet Inc.'], ['INVESTMENT', 'Person', 'Lawrence Edward Page', 'Organization', 'Kitty Hawk'], ['INVESTMENT', 'Person', 'Lawrence Edward Page', 'Organization', 'Opener'], ['INFLUENCED_BY', 'Person', 'Lawrence Edward Page', 'Person', 'Maria Montessori'], ['WORKED_AT', 'Person', 'Sergey Brin', 'Organization', 'Google'], ['INFLUENCED_BY', 'Person', 'Sergey Brin', 'Person', 'Maria Montessori']]\n"
+     ]
+    }
+   ],
+   "source": [
+    "for graph_d in graph_documents_filtered:\n",
+    "    graph = graph_d.nodes\n",
+    "    temp=[]\n",
+    "    # Access nodes\n",
+    "    for node in graph_documents_filtered[0].nodes:\n",
+    "        temp.append([node.type,node.id])  # Access node properties (e.g., entity type, labels)\n",
+    "    temp2=[]\n",
+    "    # Access relationships\n",
+    "    for relationship in graph_documents_filtered[0].relationships:\n",
+    "        temp2.append([relationship.type,relationship.source.type,relationship.source.id,relationship.target.type,relationship.target.id])\n",
```

Comment:
Is there a reason for this cell besides just printing out the relationships in the db graph?